### ISSUE TITLE
Program freezes after pasting captcha

### ISSUE THREAD

**Issue by nathom** [subdavis/kobo-book-downloader#46]
**Describe the issue**
After running `kobodl user add`, typing the credentials, and getting the captcha result, the program freezes. It won't let me press return. I tried on two different terminal apps and different shells but all of them break.

**To Reproduce**
1. Run `kobodl user add`
2. Follow instructions
3. Paste captcha
4. Program freezes

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOS 12 beta
-  Install method: Python 3.9, pip
 - Kobodl Version: 0.7.0

**Comment by nathom** [comment]
I was able to fix this by reading from the clipboard using `pyperclip` instead of getting input:

```diff
@@ -71,7 +72,9 @@ def add(ctx, email, password):
     and paste it here.  It will be very long!
     """
     )
-    captcha = input('Captcha response: ').strip()
+    input('Press enter after copying captcha').strip()
+    captcha = pyperclip.paste().strip()
+
     actions.Login(user, password, captcha)
     Globals.Settings.UserList.users.append(user)
     Globals.Settings.Save()
```

I'm still not sure whether the root cause is the OS, Terminal, Python, or something about the script though.

**Comment by otrdriver** [comment]
Same problem as reported above by Nathom. I tried different work-arounds to paste captcha results into terminal without success.

Desktop
    OS: macOS 11.4
    Install method: Python 3.8.2, pip
    Kobodl Version: 0.7.0

Stumbled here and discovered Nathom's work-around. Confirming that it works like a charm. Thanks all around.

**Comment by subdavis** [comment]
Must be an OSX issue.  I cannot reproduce this (no hardware) but I'll take you at your word and include pyperclip in the next release.

Thanks for the report.

**Comment by otrdriver** [comment]
Disclaimer: I am not a Coder.
I do however pay attention to detail, am able to follow directions and am obsessive in desire to understand why/why not something works.

Anecdotal Data: 

Gut feeling, yes likely something to do with OSX/python(?)

When I pasted the captcha result into the program the terminal bell rang on each line return/feed(?).
I understand this to indicate a bad character. I actually tried cop

If the same captcha result was pasted into a different terminal window (no python?) and used to define a variable; all worked as one would expect.

Using pyperclip does not introduce any extraneous/bad characters?

Thank you very much for your efforts. They are much appreciated.

cjs

**Comment by subdavis** [comment]
Fixed in 0.7.1 published to pypi, thanks @nathom and @otrdriver :)

**Comment by zingaburga** [comment]
Sorry if I'm hijacking an old issue, but I feel it's related.

When running on a headless Linux server, with no clipboard, I get the following crash:

	Traceback (most recent call last):
	  File "/usr/local/bin/kobodl", line 11, in <module>
	    sys.exit(cli())
	  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 829, in __call__
	    return self.main(*args, **kwargs)
	  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 782, in main
	    rv = self.invoke(ctx)
	  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 1259, in invoke
	    return _process_result(sub_ctx.command.invoke(sub_ctx))
	  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 1259, in invoke
	    return _process_result(sub_ctx.command.invoke(sub_ctx))
	  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 1066, in invoke
	    return ctx.invoke(self.callback, **ctx.params)
	  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 610, in invoke
	    return callback(*args, **kwargs)
	  File "/usr/local/lib/python3.6/dist-packages/click/decorators.py", line 33, in new_func
	    return f(get_current_context().obj, *args, **kwargs)
	  File "/usr/local/lib/python3.6/dist-packages/kobodl/commands/user.py", line 76, in add
	    captcha = pyperclip.paste().strip()
	  File "/usr/local/lib/python3.6/dist-packages/pyperclip/__init__.py", line 681, in lazy_load_stub_paste
	    return paste()
	  File "/usr/local/lib/python3.6/dist-packages/pyperclip/__init__.py", line 336, in __call__
	    raise PyperclipException(EXCEPT_MSG)
	pyperclip.PyperclipException:
	    Pyperclip could not find a copy/paste mechanism for your system.
	    For more information, please visit https://pyperclip.readthedocs.io/en/latest/index.html#not-implemented-error

I get that trying to read the clipboard was to get around terminal pasting issues, but perhaps leave it as a fallback if there's no clipboard?  
I worked around the issue by hard coding the token in the Python code, so don't know if terminal pasting would've had similar issues.

Thanks :)
